### PR TITLE
feat(flashing): prevent flashing

### PR DESCRIPTION
**Pull request by FedDiSte**
Added a functionality to prevent flashing during theme change if transition-colors is applied to elements

### CODE REVIEW

**Comment by FedDiSte** [comment]
Resolves #26

File: packages/remix-themes/src/theme-provider.tsx [hunk 1]
```diff
@@ -67,11 +95,13 @@ export function ThemeProvider({
     })
 
     broadcastThemeChange(theme)
+    disableTransitionOnThemeChange && disableTransition()
```
Comment (abereghici) [inline packages/remix-themes/src/theme-provider.tsx:98]: Hey @FedDiSte , thanks for this PR. 

`disableTransition` function seems to return a cleanup function where the styles are removed from the body. But I don't see where this cleanup function is invoked.

Comment (FedDiSte) [reply packages/remix-themes/src/theme-provider.tsx:98]: Yeah i just saw that, i forgot to remove that, i adapted the code i was using in my project, i used that function in a different way, i will commit a fix later

Comment (FedDiSte) [reply packages/remix-themes/src/theme-provider.tsx:94]: @abereghici i've just updated the correct code, sorry for the wrong commit

Comment (abereghici) [reply packages/remix-themes/src/theme-provider.tsx:94]: @FedDiSte looks good! thank you! 
I do have a question for this line `(() => window.getComputedStyle(document.body))();` why it should be wrapped in an IIFE?

Comment (FedDiSte) [reply packages/remix-themes/src/theme-provider.tsx:94]: @abereghici I used that just to force the restyle of the body, it could be outside the IIFE too

**Comment by abereghici** [comment]
@FedDiSte thank you!

**Pull request merged**